src/test/java/org/cactoos/io/LoggingInputTest.java

```diff
@@ -0,0 +1,218 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link LoggingInput}.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+@SuppressWarnings(
+    {
+        "PMD.MoreThanOneLogger",
+        "PMD.AvoidDuplicateLiterals"
+    }
+)
+public final class LoggingInputTest {
+
+    @Test
+    public void logReadFromDeadInput() throws IOException {
+        final Logger logger = new FakeLogger();
+        new LengthOf(
+            new LoggingInput(
+                new DeadInput(),
+                "dead input",
+                logger
+            )
+        ).intValue();
+        MatcherAssert.assertThat(
+            "Can't log zero byte read from dead input",
+            logger.toString(),
+            Matchers.containsString("Read 0 byte(s) from dead input in")
+        );
+    }
+
+    @Test
+    public void logReadFromOneByte() throws IOException {
+        final Logger logger = new FakeLogger();
+        new LengthOf(
+            new LoggingInput(
+                new InputOf("a"),
+                "memory",
+                logger
+            )
+        ).intValue();
+        MatcherAssert.assertThat(
+            "Can't log one byte read from memory",
+            logger.toString(),
+            Matchers.containsString("Read 1 byte(s) from memory in")
+        );
+    }
+
+    @Test
+    public void logReadFromText() throws IOException {
+        final Logger logger = new FakeLogger();
+        new LengthOf(
+            new LoggingInput(
+                new InputOf("Hello, товарищ!"),
+                "memory",
+                logger
+            )
+        ).intValue();
+        MatcherAssert.assertThat(
+            "Can't log 22 bytes read from memory",
+            logger.toString(),
+            Matchers.containsString("Read 22 byte(s) from memory in")
+        );
+    }
+
+    @Test
+    public void logReadFromLargeTextFile() throws IOException {
+        final Logger logger = new FakeLogger();
+        new LengthOf(
+            new LoggingInput(
+                new ResourceOf("org/cactoos/large-text.txt"),
+                "text file",
+                logger
+            )
+        ).intValue();
+        MatcherAssert.assertThat(
+            "Can't log 74536 bytes read from text file",
+            logger.toString(),
+            Matchers.allOf(
+                Matchers.not(
```

Comment:
@fabriciofx you need to add more conditions here because you use ``containsString``, so you need to be sure that "Read 74536 byte(s) from text file in" is the only content. Why don't you use ``equalTo`` and have only one assertion?

Reply:
@amihaiemil I'm using `containsString` because of the time in the of the message. If the test will ne executed in a more slow machine, the time will change and the test will fail. `equalTo` test the same string, but they're not equal, because has the time. The reason of two assertion is because the test can be match with the detailed case (WARNING). So this case is to match one only message.